exchange_calendars/calendar_utils.py

```diff
@@ -196,24 +196,32 @@ def get_calendar(self, name):
         calendar = self._calendars[canonical_name] = factory()
         return calendar
 
-    def get_calendar_names(self):
-        """
-        Returns all the calendars we know about or know how to make
+    def get_calendar_names(self, include_aliases: bool = True) -> list[str]:
+        """Return all canoncial calendar names and, optionally, aliases.
+
+        Parameters
+        ----------
+        include_aliases : default: True
+            True to include calendar aliases.
+            False to return only canonical calendar names.
 
         Returns
         -------
-        calendar_names: List[str]
-            A list of all the calendars we know about or know how to make
+        list of str
+            List of canonical calendar names and, optionally, aliases.
+
+        See Also
+        --------
+        names_to_aliases : Mapping of cononcial names to aliases.
+        aliases_to_names : Mapping of aliases to canoncial names.
+        resolve_alias : Resolve single alias to a canonical name.
         """
-        return list(
-            set(
-                itertools.chain(
-                    self._calendars.keys(),
-                    self._calendar_factories.keys(),
-                    self._aliases.keys(),
-                )
-            )
-        )
+        keys = set(self._calendar_factories.keys()).union(set(self._calendars.keys()))
+        if include_aliases:
+            keys = keys.union(set(self._aliases.keys()))
+        names = list(keys)
+        names.sort()
```

Comment:
Any reason we actually need to sort here?

For any reasonable amount of calendars this will never be a bottleneck, but wanted to check if we depend on this.

Reply:
Thanks for pointing this out. The sort was only to make the return friendlier for users querying calendar names. I've added a `sort` option, default True, and passed it as False from the only internal caller `resolve_alias` (which in turn is called from `get_calendar`).